File: firebase/firestore/lib/src/faculty/reader.dart
import "package:firestore/data.dart";
import "package:firestore/helpers.dart";

/// A collection of functions to read faculty data.
/// 
/// No function in this class actually performs logic on said data, just returns
/// it. This helps keep the program modular, by separating the data sources from
/// the data indexing.
class FacultyReader {
	/// Maps faculty IDs to their respective [User] objects.
	static Future<Map<String, User>> getFaculty() async => {
		await for(final Map<String, dynamic> row in csvReader(DataFiles.faculty))
			row ["ID"]: User(
				id: row ["ID"],
				email: row ["E-mail"].toLowerCase(),
				first: row ["First Name"],
				last: row ["Last Name"],
			)
	};
}

File: firebase/firestore-py/lib/faculty/reader.py
import csv
from ..utils import dir
from ..data import student

'''
A collection of functions to read faculty data.
No function in this class actually performs logic on said data, just returns
it. This helps keep the program modular, by separating the data sources from
the data indexing.
'''
def get_faculty():
  with open(dir.faculty) as file:
    return {row["USER_ID"]: student.User(
            first = row["FIRST_NAME"],
            last = row ["LAST_NAME"],
            email = row["EMAIL"].lower(),
            id = row["USER_ID"])
            for row in csv.DictReader(file)}

